setup.py

```diff
@@ -58,37 +48,95 @@ def get_setup_version(reponame):
         print("WARNING: param>=1.6.0 unavailable. If you are installing a package, this warning can safely be ignored. If you are creating a package or otherwise operating in a git repository, you should install param>=1.6.0.")
         return json.load(open(version_file_path, 'r'))['version_string']
 
-setup_args.update(dict(
+
+####################
+### dependencies ###
+
+# TODO: versions need to be updated
+install_requires = [
+    'bokeh >=0.12.13',   # not strictly required but shouldn't be problematic
+    'cartopy >=0.14.2',  # prevents pip alone (requires external package manager)
+    'holoviews >=1.9.4',
+    'numpy >=1.0',
+    'param >=1.5.1',
+]
+
+
+# TODO: need to verify this list (not sure all needed), plus some
+# versions may be needed
+extras_require={
+    'recommended': [
+        'filelock', # TODO for iris?
+        'datashader',
+        'geopandas',
+        'gdal', 'libgdal',
+        'matplotlib',
+        'pandas',
+        'scipy',
+        'shapely',
+        'xarray',
+        #### intent of packages below is jupyter notebook with ipykernel?
+        # seems like it could be simplified (or if for another
+        # purpose, should be clarified)
```

Comment:
That's right, I don't think all those are needed, ``notebook`` is probably sufficient.

Reply:
It's probably notebook and ipykernel, or something like that. I've never tried to figure it out exactly...